### PR TITLE
Add KnownOperationTypes rule

### PR DESCRIPTION
**Pull request by yaacovCR**
New Spec PR: https://github.com/graphql/graphql-spec/pull/1098
Old Spec PR: https://github.com/graphql/graphql-spec/pull/947
Original issue raised by @benjaminjkraft : #3592

### CODE REVIEW

**Comment by yaacovCR** [comment]
@github-actions publish-pr-on-npm

File: src/validation/__tests__/KnownOperationTypesRules-test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  });
+
+  it('mixture of known and unknown operations', () => {
+    expectErrors(`
+      query {
+        field
+      }
+      mutation {
+        field
+      }
```
Comment (IvanGoncharov) [inline src/validation/__tests__/KnownOperationTypesRules-test.ts:44]: nit: in the case of dummy fields, you can use single line notation for each operation.
```
query { field }
mutation { field }
```

Comment (yaacovCR) [reply src/validation/__tests__/KnownOperationTypesRules-test.ts:44]: This style was taken from existing rules.

Comment (yaacovCR) [reply src/validation/__tests__/KnownOperationTypesRules-test.ts]: ah, likely those files have other selection sets with more than one field and better to be consistent across the file.

Comment (yaacovCR) [reply src/validation/__tests__/KnownOperationTypesRules-test.ts]: https://github.com/graphql/graphql-js/commit/36163fbffde623cac962ab7898c26ec276ae2a53

**Comment by benjaminjkraft** [comment]
Neat, thanks!

File: src/validation/__tests__/KnownOperationTypesRules-test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    `).toDeepEqual([
+      {
+        message: 'The mutation operation is not supported by the schema.',
+        locations: [{ line: 2, column: 7 }],
+      },
+    ]);
+  });
+
```
[the remark this answers is not in the document]
Comment (yaacovCR) [reply src/validation/__tests__/KnownOperationTypesRules-test.ts:32]: Makes sense! https://github.com/graphql/graphql-js/pull/3601/commits/027c056fe94191e17602e27c02f17af0fa2de27b

File: src/validation/index.ts [hunk 3]
```diff
@@ -24,6 +24,9 @@ export { KnownDirectivesRule } from './rules/KnownDirectivesRule';
 // Spec Section: "Fragment spread target defined"
 export { KnownFragmentNamesRule } from './rules/KnownFragmentNamesRule';
 
+// Spec Section: "Operation Type Existence"
+export { KnownOperationTypesRule } from './rules/KnownOperationTypesRule';
+
```
[the remark this answers is not in the document]
Comment (yaacovCR) [reply src/validation/index.ts:29]: Yup, an error I have made before and will make again. :( Thanks! https://github.com/graphql/graphql-js/pull/3601/commits/b3cc8f413c540cf6c1de3d4ab53b3a0c714287cb

**Review by IvanGoncharov — APPROVED** [review]
Looks good now 👍 
So let's wait for stage2 to merge it.
Personally, I'm for fast-forwarding it to stage2 the next wg.

**Review by saihaj — APPROVED** [review]

**Comment by benjaminjkraft** [comment]
For the sake of the record, the spec PR is now graphql/graphql-spec#955 and is scheduled for discussion on Thursday.

**Comment by yaacovCR** [comment]
There is a new spec PR https://github.com/graphql/graphql-spec/pull/1098 originally created by @Shane32 and championed at the October 2024 WG meeting by @benjie

**Review by Shane32 — APPROVED** [review]

**Comment by benjie** [comment]
We're back to:

- https://github.com/graphql/graphql-spec/pull/955

I'm happy to continue to champion this at the WG unless someone else would like to take it on.

**Comment by yaacovCR** [comment]
This advanced to [Stage 2 at the WG meeting from 12/5/2024](https://github.com/graphql/graphql-wg/blob/main/agendas/2024/12-Dec/05-wg-primary.md) => so merging to at least v17.

**Pull request merged**